### ISSUE TITLE
Default TLS Curves?

### ISSUE THREAD

**Issue by DylanYoung** [openssl/openssl#11548]
Question and possible feature enhancement:

Nginx docs for the `ssl_ecdh_curve` option states the following:

> The special value auto (1.11.0) instructs nginx to use a list built into the OpenSSL library when using OpenSSL 1.0.2 or higher, or prime256v1 with older versions.

Which is all well and good, but I've been unable to turn how to find what this default is. 

Is there an openssl command to run to show the defaults (similar to the `ciphers` command)?

If not, can you direct me to where in the source these are set?  Can we get a command to output the defaults so we don't have to browse the source for each new version? 

Apologies if this is obvious, but I can't seem to find it. 

Note that I'm not looking for an answer like "these are the current defaults" as we need to be able to maintain out systems going forward.

**Comment by richsalz** [comment]
You want to know which curves are supported by default in TLS?  I assume you want something like this? https://github.com/openssl/openssl/blob/master/ssl/t1_lib.c#L195

**Comment by kroeckx** [comment]
I strongly suggest that you keep the auto setting and do not maintain your own settings.

**Comment by richsalz** [comment]
I agree with Kurt's recommendation, I was just curious what they were asking for.

**Comment by DylanYoung** [comment]
@kroeckx Wasn't intending to change it, but we hold liability for the security of our application, so we need to know the level of that security at any given time.

Any chance of getting a method of outputting this info?

**Comment by DylanYoung** [comment]
@richsalz  Thanks!

**Comment by DylanYoung** [comment]
Not to mention general curiosity of course :)

**Comment by DylanYoung** [comment]
I'd be happy to take a crack at an outputting patch if you think it'd be accepted.  Would `openssl-ecparam` be a good place for this?

**Comment by DylanYoung** [comment]
The one thing that does pop out to me is that the defaults seem to be generally ordered in the reverse order from what I would expect for setup favouring security over speed.  You can see that many people agree on this based on the number of posts/sites that recommend switching the order up.  Obviously it makes zero impact to most common use-cases, but still...

**Comment by kroeckx** [comment]
I assume that SSL_get1_curves() returns that list.

Setting the security level higher will make it stop using things
from that list that doesn't match the security level.

The order depends on more than just the security of the curve
and the speed, but for instance also on the quality of the
implementation.

**Comment by mattcaswell** [comment]
> SSL_get1_curves()

An annoying quirk of the OpenSSL API is that `SSL_set1_curves()` sets *our* curves, but `SSL_get1_curves` gets the curves sent by the peer.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
